src/tab/Tabs.ts

```diff
@@ -0,0 +1,259 @@
+ */
+@customElement('omni-tabs')
+export class Tabs extends OmniElement {
+    @state() private _observer: MutationObserver | undefined;
+
+    override connectedCallback(): void {
+        super.connectedCallback();
+        this._observer = new MutationObserver((mutations) => {
+            for (const mutation of mutations) {
+                if (mutation.type === `attributes`) {
+                    // Re-render the component when a child item's attributes has changed.
+                    this.requestUpdate();
+                }
+            }
+
+            // Start observing child attribute changes.
+            this._observer?.observe(this, {
+                attributes: true,
+                attributeFilter: [`header`],
+                subtree: true
+            });
+        });
+    }
+
+    /**
+     * Clean-up the component once removed from the DOM.
+     *
+     * @ignore
+     *
+     * @returns {void}
+     */
+    override disconnectedCallback() {
+        // Stop observing child attribute changes.
+        if (this._observer) {
+            this._observer.disconnect();
+        }
+
+        // Ensure the component is cleaned up correctly.
+        super.disconnectedCallback();
+    }
+
+    selectTab(tabHeader: TabHeader) {
+        if (!tabHeader || tabHeader.classList.contains('tab-bar')) {
+            return;
+        }
+
+        // set tab header used in cases where the tab header has slotted content.
+        tabHeader = tabHeader.closest('omni-tab-header') as TabHeader;
+        const children = Array.from(this.children);
+
+        const tab = children.find((t) => (t.id && t.id === tabHeader.for) || t === tabHeader.data);
+
+        if (!tab || tab.hasAttribute(disabledAttribute)) {
+            return;
+        }
+
+        const tabHeaders = [
+            ...children.filter((oth) => oth.slot === 'header'),
+            ...(this.shadowRoot?.querySelector<HTMLSlotElement>('slot[name=header]')?.children || [])
+        ] as TabHeader[];
+
+        const previous = children.find((c) => c.hasAttribute(activeAttribute));
+
+        // Remove active attributes from tab headers and tabs.
+        tabHeaders.forEach((header) => {
+            header.removeAttribute(activeHeaderAttribute);
+            header.requestUpdate();
+        });
+        children.forEach((element) => {
+            element.removeAttribute(activeAttribute);
+        });
+
+        // Set active tab-header and tab
+        tab.setAttribute(activeAttribute, '');
+        tabHeader.setAttribute(activeHeaderAttribute, '');
+        tabHeader.requestUpdate();
+
+        this.dispatchEvent(
+            new CustomEvent('tab-select', {
+                detail: {
+                    previous: previous,
+                    selected: tab
+                }
+            })
+        );
+
+        this.requestUpdate();
+    }
+
+    static override get styles() {
+        return [
+            super.styles,
+            css`
+            :host {
```

Comment:
Check code indentation inside the `css` template function.

Reply:
Updated